Tensile/TensileCreateLibrary.py

```diff
@@ -266,6 +266,42 @@ def prepAsm():
   assemblerFile.close()
   os.chmod(assemblerFileName, 0o777)
 
+################################################################################
+def prepAsmNewClient(kernelWriterAssembly):
+  """
+  Create and prepare the assembly directory  - called ONCE per output dir:
+  """
+  asmPath = ensurePath(os.path.join(globalParameters["WorkingPath"], "assembly") )
+  assemblerFileName = os.path.join(asmPath, \
+      "asm-new.%s"%("bat" if os.name=="nt" else "sh"))
+  assemblerFile = open(assemblerFileName, "w")
+  if os.name == "nt":
+    assemblerFile.write("echo Windows: Copying instead of Assembling\n")
+    assemblerFile.write("copy %1.s %1.o\n")
+    assemblerFile.write("copy %1.o %1.co\n")
+  else:
+    assemblerFile.write("#!/bin/sh %s\n" % ("-x" if globalParameters["PrintLevel"] >=2  else ""))
+    assemblerFile.write("# usage: asm-new.sh kernelName(no extension)\n")
+
+    assemblerFile.write("f=$1\n")
+    assemblerFile.write("shift\n")
+
+    isa = globalParameters["CurrentISA"]
+    assemblerFile.write("h=gfx%s\n" % "".join(map(str,isa)))
```

Comment:
- Should use `Common.gfxName()` to convert this to the architecture.
 - I'd like to try to migrate to using `str.format()` instead of `%` formatting.

Reply:
- Useful function to know about
- Agreed on migrating to `str.format()`.